app/src/main/java/org/mozilla/fenix/home/sessioncontrol/SessionControlController.kt

```diff
@@ -561,4 +569,35 @@ class DefaultSessionControlController(
         metrics.track(Event.CloseExperimentCardClicked)
         fragmentStore.dispatch(HomeFragmentAction.RemoveSetDefaultBrowserCard)
     }
+
+    override fun handlePrivateModeButtonClicked(
+        newMode: BrowsingMode,
+        userHasBeenOnboarded: Boolean
+    ) {
+        val isPrivateMode = newMode == BrowsingMode.Private
+        if (isPrivateMode) {
+            activity.settings().incrementNumTimesPrivateModeOpened()
+        }
+
+        if (userHasBeenOnboarded) {
+            fragmentStore.dispatch(
+                HomeFragmentAction.ModeChange(Mode.fromBrowsingMode(newMode))
+            )
+        }
+
+        // If we are changing the mode using the homescreen mask toggle
+        // from behind the search dialog, open the url in a new tab with the new mode
+        val fromHome = navController.previousBackStackEntry?.destination?.id == R.id.homeFragment
+        val onSearch = navController.currentDestination?.id == R.id.searchDialogFragment
+        val stillOnATab = store.state.selectedTab?.engineState?.engineSession != null
+
+        if (fromHome && onSearch && stillOnATab) {
+            store.state.selectedTab?.content?.url?.let { url ->
```

Comment:
We shouldn't copy the `content.url` of another tab here esp. if we're switching from private to normal mode. I know this isn't loading the tab yet, but we should also not copy the URL or any other state.

It should be good enough to do this and initialize the new tab with `about:blank`:
```kotlin
if (store.state.selectedTab != null) {
  addTabUseCase.invoke(
    private = isPrivateMode
  )
}
```

Reply:
Ahh that totally makes sense. I didn't realize we could do this without having the URL 👍